data-service/src/main/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDB.java

```diff
@@ -396,25 +395,36 @@ public void deleteEnvironmentRevision(@NonNull final EnvironmentRevision environ
       Validate.notNull(accountId, "accountId must be specified if clusterNamePrefix is specified");
     }
 
-    // TODO: rewrite this to use the index instead of scanning the table
-    PaginatedScanList<EnvironmentDDBRecord> records =
-        dynamoDBMapper.scan(EnvironmentDDBRecord.class, new DynamoDBScanExpression());
+    PaginatedList<EnvironmentDDBRecord> clusters;
 
-    Stream<Cluster> stream =
-        records
-            .stream()
-            .map(EnvironmentDDBRecord::getAccountIdCluster)
-            .distinct()
-            .map(Cluster::fromAccountIdCluster);
+    if (accountId == null) {
+      clusters =
+          dynamoDBMapper.scan(
+              EnvironmentDDBRecord.class,
+              new DynamoDBScanExpression()
+                  .withIndexName(EnvironmentDDBRecord.ENVIRONMENT_CLUSTER_GSI_NAME));
+    } else {
+      DynamoDBQueryExpression<EnvironmentDDBRecord> query =
+          new DynamoDBQueryExpression<EnvironmentDDBRecord>()
+              .withIndexName(EnvironmentDDBRecord.ENVIRONMENT_CLUSTER_GSI_NAME)
+              .withHashKeyValues(EnvironmentDDBRecord.builder().accountId(accountId).build())
+              .withConsistentRead(false);
```

Comment:
Dynamo GSI don't support consistent read,  not sure if this configuration is necessary.

Reply:
It is; the default is `true`, and if this is omitted, the call fails.